### DISCUSSION TITLE
Help needed instancing using PointInstancer node and Prototypes

### DISCUSSION THREAD

**Discussion by andy3dcouk** [Autodesk/maya-usd#1355] [Q&A]
Hi Everyone,
I'm trying to get started with USD in Maya 2022 to see if it could help me populate a 50,000 seater baseball stadium I'm building. I'm wanting a single seat to be instanced across the entire stadium for all the obvious reasons (speed/ease to edit seat geom etc).

I'm hoping my scene can be very light (like Pixars 'PointInstancedMedCity' example usd file) but I can't find any information on how to create this 'PoinInstancer >> Prototype' relationship. All I'm managing to do so far is create a 1GB usd file with 50,000 seat xforms/shapes positioned to each seat locator via a simple python script. 

It's amazingly fast to traverse the scene or render with this usd file added to a stage, but I don't feel this is really using USD correctly.

Can anyone give any basic information on how to get started with USD PointInstancer and Prototypes in Maya?

THANKS IN ADVANCE! 
Andy

**Comment by ppt-adsk** [comment]
Hi @andy3dcouk, unfortunately we don't have any tools currently in Maya to help you author point instancers and their instances.  However, if you're comfortable with Python scripting (and it seems you are), you can create your scene with a point instancer and its single prototype, creating as many point instances as you like, and you can use Maya to edit the resulting point instancer and its instances.

Have a look at these test files:

https://github.com/Autodesk/maya-usd/tree/dev/test/testSamples/pointInstances

I'm confident that if you use these as inspiration you should be on your way.

**Reply by andy3dcouk** [reply]
Thanks @ppt-adsk
I'm using USD 0.9 in Maya 2022. I can't 'Import' or 'create stage from file' for any of the pointInstances samples. I can't even use Hierarchy view during import. Errors out what ever I try.

Are these samples I'm trying to open going to show me that I CAN point instance usd geom within Maya?

**Reply by andy3dcouk** [reply]
I've managed to open the samples in Maya now, and also had a good look through the usda files in a text editor. 

But I can only see how I'd go about doing something similar in raw text and saving as usda, not in the Maya UI. 

When you say "_unfortunately we don't have any tools currently in Maya to help you author point instancers and their instances_" is that what you mean?

**Reply by ppt-adsk** [reply]
That's right.  We don't have any workflows to create point instancers, their prototypes, and their point instances, but if you can do this outside of Maya, you can bring them into Maya natively using "create stage from file", then edit the point instances, e.g. you can use move, rotate, scale on them.

**Reply by andy3dcouk** [reply]
OK Thanks for clearing that up. Is there an app recommended for creating these native usd nodes? Eg. How would the Pixar Med City been created?

**Reply by sylvieChen** [reply]
I think we can use Maya Mash. just use the mayaUsdProxyShape node as prototype, then use the USD exporter to export the mash node to USD. Then it will generate point instancer sturcture for you. 

but I found that when I use AL_usdmaya_ProxyShape node as prototype, the prototype would be a reference in the exported USD file. but when I use mayaUsdProxyShape node as prototype, there is no reference in the prototype. 
Also I found I don't see the instanced object in either vp2 or hydra viewport.

**Reply by cgTD123** [reply]
> That's right. We don't have any workflows to create point instancers, their prototypes, and their point instances, but if you can do this outside of Maya, you can bring them into Maya natively using "create stage from file", then edit the point instances, e.g. you can use move, rotate, scale on them.

I bring pointInstance usd using "create stage from file", but how to edit the poing instances,all attribute can't edited,and the point can't be selected
